File: fatorial/fat.c
#include <stdio.h>

int main () {

    int n, n2, fat;

    printf("Digite um número: ");
    scanf("%d", &n);

    n2 = n;
    fat = 1;

    for (size_t c = n; c > 0; c--) {
        fat = fat * n;
        n--;
    }
    
    printf("%d! = %d", n2, fat);

    return 0;
}

File: fatorial/fat.java
import java.util.Scanner;

class Main {  
    public static void main(String args[]) {  
        Scanner input = new Scanner(System.in);
        System.out.println("Digite um numero: ");
        int n = input.nextInt();
        input.close();
        int n2 = n;
        int fat = 1;

        for (int c = n; c > 0; c--) {
            fat = fat * n;
            n--;
        }

        System.out.printf("%d! = %d", n2, fat);
    }  
}